File: edge.js
    let arr = [1,3,5,6,2];
    lo = 0;
    hi = arr.length;    
    let key = prompt('enter key');
    key = parseInt(key);
    function rank(key,arr){
        while(lo <= hi)
        {
            mid = lo + (hi - lo) / 2;
            if (key < arr[mid]) {
                hi = mid - 1;
            } else if (key > arr[mid]) {
                lo = mid + 1;
            } else {
                return mid;
            }
        }
        return -1;
    }
rank(key,arr);

File: edge.py

key = input("enter your value: ")
a = [1,2,3,4,5]
def rank(key, a):
    lo = 0
    hi = len(a)
    while lo <= hi:
        mid = lo + (hi - lo) / 2
        key = 2
        if key < a[mid]:
            hi = mid - 1
        elif key > a[mid]:
            lo = mid + 1
        else :
            return mid
        return - 1

